zirgen/circuit/keccak/README.md

```diff
@@ -0,0 +1,52 @@
+
+### Keccak Input Transcript
+
+| Name          | Byte Length | Description                                                          |
+|---------------|-------------|----------------------------------------------------------------------|
+| data blocks   | arbitrary   | raw data blocks containing the block count followed by the raw data. |
+| final padding |           8 | Padding used to indicate the end of the keccak input transcript.     |
+
+### Data Blocks
+
+Data blocks represent the raw data being hashed. This contains an 2-byte value that represents the block size followed by a sequence of bytes representing the raw data, ending with the output digest generated by running keccak.
+
+| Name             | Byte Length | Description                                                                                 |
+|------------------|-------------|---------------------------------------------------------------------------------------------|
+| block count (BC) |          2  | the amount of blocks that the raw data spans. Each block is 136 bytes.                      |
+| padding          |          6  | Padding containing all 0 values. This is used to align the length along an 8-byte boundary. |
+| raw data         |   BC * 136  | Represents the raw data.                                                                    |
+| output digest    |         32  | the keccak hash of the raw data.                                                            |
+
+### Example: using the accelerator in Rust crates
+
+Most rust cryptographic hashing crates use an API similar to the following code snippet:
+```
+let mut hasher = Keccak256::new();
+
+// write input message
+hasher.update(&input);
+
+// read hash digest
+let result = hasher.finalize();
+```
+
+In this code snippet, `hasher` carries an internal state which is modified the input using the `update(input)` function. `finalize` will generate a resulting hash. In order to use the accelerator we need to generate an input transcript. To keep the proof under 2 million cycles the maximum amount of data we can hash in the accelerator is approximately 95 kilobytes.
+#### WIP: Non-concurrent implementation for crates such as tiny-keccak and sha3
+
+The following assumes a situation where a single call to finalize is compressing less than 95 KB of input data and only one instance of hasher is present.
+
+* Every time update is called, the input is split up to fit inside the 136 byte block. Once the block fills up, it is compressed using keccak. In our design, we send each of the blocks to the host in order to record the raw data of the input transcript.
+* Once the hasher's `finalize` function is called we can communicate with the termination so that the block count can be computed and for the keccak hash to be generated.
+
+#### WIP: Non-concurrent implementation for crates such as sha3
+
+#### WIP: concurrent implementation
```

Comment:
@iddo-bentov can you remind me what we've decided to do here? is this a WIP for the nearterm or something we've decided to punt on for now? 

might make sense to strike this section from the public readme and update our linear board to reflect the work we have planned here.

Reply:
Indeed there's no need for the WIP comments to be in the public readme.
The current version is also the most efficient version nearterm, we will assess the concurrent version after we finish integrating this version.